src/main/java/com/synopsys/integration/jenkins/detect/service/DetectEnvironmentService.java

```diff
@@ -66,10 +68,14 @@ private void populateAllBlackDuckEnvironmentVariables(BiConsumer<String, String>
         BlackDuckServerConfigBuilder blackDuckServerConfigBuilder = detectGlobalConfig.get().getBlackDuckServerConfigBuilder(jenkinsProxyHelper, synopsysCredentialsHelper);
 
         blackDuckServerConfigBuilder.getProperties()
-            .forEach((builderPropertyKey, propertyValue) -> acceptIfNotNull(environmentPutter, builderPropertyKey.getKey(), propertyValue));
+            .forEach((builderPropertyKey, propertyValue) -> updateAndFilterVariables(environmentPutter, builderPropertyKey.getKey(), propertyValue));
     }
 
-    private void acceptIfNotNull(BiConsumer<String, String> environmentPutter, String key, String value) {
+    private void updateAndFilterVariables(BiConsumer<String, String> environmentPutter, String key, String value) {
+        if ("BLACKDUCK_TIMEOUT".equals(key)) {
+            key = TIMEOUT;
+        }
+
         if (StringUtils.isNoneBlank(key, value)) {
             environmentPutter.accept(key, value);
         }
```

Comment:
Generally, mutating a method parameter is considered to be bad form unless you make it very clear that you're going to be mutating that parameter. I think we on the team would prefer:

```suggestion
        String filteredKey = key;
        if ("BLACKDUCK_TIMEOUT".equals(filteredKey)) {
            filteredKey = TIMEOUT;
        }

        if (StringUtils.isNoneBlank(filteredKey, value)) {
            environmentPutter.accept(filteredKey, value);
        }
```

Though I'd be curious if @ekerwin would tell me to stop whining =)

Reply:
PR updated